File: bj/14626.cpp
#include<iostream>
#include<string>
#include<vector>
using namespace std;

int main(){
    string isbn;
    cin >> isbn;
    vector<int> weight = {1,3,1,3,1,3,1,3,1,3,1,3,1};
    int checksum = 0; 
    int masked = 0;
    for (int i = 0; i < isbn.length(); i++){
        if (isbn[i] != '*') {
            checksum += (isbn[i]-'0')*weight[i];
        }
        else{
            masked = i;
        }
    }
    for (int j = 0; j < 10; j++){
        if (((weight[masked]*j + checksum)%10) == 0) {
            cout << j;
            break;
        }
    }
    return 0;
}

File: bj/14626.py
import sys
isbn = str(sys.stdin.readline().strip())
weight = [1,3,1,3,1,3,1,3,1,3,1,3,1]
checksum = 0
masked = 0
for i in range(len(isbn)):
    if isbn[i] != '*':
        checksum += int(isbn[i])*weight[i]
    else:
        masked = i

for j in range(10):
    if (weight[masked]*j + checksum)%10 == 0:
        print(j)
        break